webui/react/src/ioTypes.ts

```diff
@@ -44,6 +44,32 @@ export class ValueofType<D extends { [key: string]: unknown }> extends io.Type<V
   }
 }
 
+export class Float extends io.Type<number> {
+  readonly _tag: 'FloatType' = 'FloatType' as const;
+  constructor() {
+    super(
+      'float',
+      (u): u is number => io.number.is(u),
+      (u, c) => {
+        let u_ = u;
+        if (u === 'Infinity') {
+          u_ = Infinity;
+        }
+        if (u === '-Infinity') {
+          u_ = -Infinity;
+        }
+        if (u === 'NaN') {
+          u_ = NaN;
+        }
+        return io.number.validate(u_, c);
+      },
+      (f) => {
+        return io.number.encode(f);
```

Comment:
You need to reverse the transformation you made above in the encoder. `JSON.stringify` does not convert NaN, etc to the strings you've put above.

Reply:
That's where I got confused.. Since we extend `io.Type<number>` and that's `Type<number, number, unknown>`, the encode function should only return `number`?

Reply:
Just a typo on my part. Generally we want both the input and output types to be `unknown` so that should be `Type<number, unknown, unknown>`

Reply:
Or you can make it `Type<number, number | string, unknown>` since that will work for JSON cases as well. Really we should define a proper `Json` type at some point.